components/CommandManager/CommandManager/commands/device_commands.cpp

```diff
@@ -154,6 +144,51 @@ CommandResult switchModeCommand(std::shared_ptr<DependencyRegistry> registry, co
     return CommandResult::getSuccessResult("Device mode switched, restart to apply");
 }
 
+CommandResult switchModeAndRestartCommand(std::shared_ptr<DependencyRegistry> registry, const nlohmann::json &json)
+{
+    if (!json.contains("mode") || !json["mode"].is_string())
+    {
+        return CommandResult::getErrorResult("Invalid payload - missing mode");
+    }
+
+    auto modeStr = json["mode"].get<std::string>();
+    StreamingMode newMode;
+
+    ESP_LOGI("[DEVICE_COMMANDS]", "Switch mode and restart command received with mode: %s", modeStr.c_str());
+
+    if (modeStr == "uvc")
+    {
+        newMode = StreamingMode::UVC;
+    }
+    else if (modeStr == "wifi")
+    {
+        newMode = StreamingMode::WIFI;
+    }
+    else if (modeStr == "setup" || modeStr == "auto")
+    {
+        newMode = StreamingMode::SETUP;
+    }
+    else
+    {
+        return CommandResult::getErrorResult("Invalid mode - use 'uvc', 'wifi', or 'auto'");
+    }
+
+    const auto projectConfig = registry->resolve<ProjectConfig>(DependencyType::project_config);
+    StreamingMode currentMode = projectConfig->getDeviceMode();
+
+    if (currentMode == newMode)
+    {
+        ESP_LOGI("[DEVICE_COMMANDS]", "Device already in mode: %d, no restart needed", (int)currentMode);
+        return CommandResult::getSuccessResult("Device already in requested mode");
+    }
+
+    ESP_LOGI("[DEVICE_COMMANDS]", "Setting device mode to: %d and scheduling restart", (int)newMode);
+    projectConfig->setDeviceMode(newMode);
```

Comment:
I get the idea behind this command and why it's needed, I completely agree with that. But do we have to duplicate the already existing `switchModeCommand()`? Is there something preventing us from just calling that command in here and then scheduling the restart right away, while returning what `switchModeCommand` returned?